packages/vm/test/api/t8ntool/t8ntool.spec.ts

```diff
@@ -34,11 +47,209 @@ const args: T8NOptions = {
 describe('test runner config tests', () => {
   it('should run t8ntool with inputs and report the expected output', async () => {
     await TransitionTool.run(args)
-    const expectedResult = JSON.parse(readFileSync(`${t8nDir}/output/result.json`).toString())
-    const expectedAlloc = JSON.parse(readFileSync(`${t8nDir}/output/alloc.json`).toString())
-    const reportedResult = JSON.parse(readFileSync(`${t8nDir}/output/resultTEST.json`).toString())
-    const reportedAlloc = JSON.parse(readFileSync(`${t8nDir}/output/allocTEST.json`).toString())
+    const expectedResult = JSON.parse(readFileSync(`${t8nDir}output/result.json`).toString())
+    const expectedAlloc = JSON.parse(readFileSync(`${t8nDir}output/alloc.json`).toString())
+    const reportedResult = JSON.parse(readFileSync(`${t8nDir}output/resultTEST.json`).toString())
+    const reportedAlloc = JSON.parse(readFileSync(`${t8nDir}output/allocTEST.json`).toString())
     assert.deepStrictEqual(reportedResult, expectedResult, 'result matches expected result')
     assert.deepStrictEqual(reportedAlloc, expectedAlloc, 'alloc matches expected alloc')
   })
 })
+describe('trace tests', async () => {
+  it('should produce a valid step trace for a legacy contract', async () => {
+    const common = new Common({ chain: Mainnet, hardfork: Hardfork.Cancun })
+    const sm = new MerkleStateManager({ common })
+    const vm = await createVM({ common, stateManager: sm })
+    const bytecode = hexToBytes('0x604260005260206000F3') // PUSH1 42 PUSH1 00 MSTORE PUSH1 20 PUSH1 00 RETURN
+    const contractAddress = createAddressFromPrivateKey(randomBytes(32))
+    await vm.stateManager.putAccount(contractAddress)
+    await vm.stateManager.putCode(contractAddress, bytecode)
+    const trace: string[] = []
+    vm.evm.events!.on('step', (step) => {
+      trace.push(JSON.stringify(stepTraceJSON(step, true)))
+    })
+    vm.events!.on('afterTx', async (event) => {
+      trace.push(JSON.stringify(await summaryTraceJSON(event, vm)))
+    })
+    const tx = await createTx({
+      to: contractAddress,
+      data: bytecode,
+      gasLimit: 0xffffffff,
+      gasPrice: 0xf,
+    }).sign(randomBytes(32))
+    await runTx(vm, { tx, skipBalance: true, skipBlockGasLimitValidation: true, skipNonce: true })
+    assert.equal(trace.length, 7, 'trace length is 7')
+    assert.equal(JSON.parse(trace[6]).gasUsed, 21154)
+  })
+  it('should produce a trace of the correct length', async () => {
+    const common = new Common({
+      hardfork: Hardfork.Prague,
+      eips: [663, 3540, 3670, 4200, 4750, 5450, 6206, 7069, 7480, 7620, 7692, 7698],
```

Comment:
This should technically be our current definition of Osaka fork right?

Reply:
Indeed.